Sources/PIALibrary/Common/Data/Networking/NetworkRequestClient.swift

```diff
@@ -43,6 +43,11 @@ private extension NetworkRequestClient {
     /// Serial execution of all the connections until one succeeds or completes with an error when all connection attempts fail
     func executeRecursivelyUntilSuccess(connections:  [NWHttpConnectionType], completion: @escaping NetworkRequestClientType.Completion) {
         
+        guard !connections.isEmpty else {
+            completion(.allConnectionAttemptsFailed(statusCode: nil), nil)
+            return
+        }
+        
```

Comment:
I added this check here because if we call this method with an empty array of connections, then the app will crash